### PR TITLE
Services/FS: Implemented DeleteExtSaveData, CreateSystemSaveData and Del...

### PR DESCRIPTION
**Pull request by Subv**
...eteSystemSaveData

Also fixed a bug with CreateExtSaveData that made it unable to create ExtSaveData archives in the SDMC directory.

This allows Nintendo Zone to boot further

### CODE REVIEW

File: src/core/hle/service/fs/fs_user.cpp [hunk 1]
```diff
@@ -520,6 +540,48 @@ static void CardSlotIsInserted(Service::Interface* self) {
...
+ *  Outputs:
+ *      1 : Result of function, 0 on success, otherwise error code
+ */
+static void CreateSystemSaveData(Service::Interface* self) {
+    // TODO(Subv): Figure out the other parameters.
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 savedata_high = cmd_buff[1];
+    u32 savedata_low = cmd_buff[2];
+
+    cmd_buff[1] = CreateSystemSaveData(savedata_high, savedata_low).raw;
```
Comment (bunnei) [inline src/core/hle/service/fs/fs_user.cpp]: Given that there are so many parameters that are unknown, I'd presume that a warning log (for unimplemented functionality) would be fitting here....

File: src/core/hle/service/fs/fs_user.cpp [hunk 2]
```diff
@@ -490,25 +490,45 @@ static void FormatThisUserSaveData(Service::Interface* self) {
...
+ *  Inputs:
+ *      0 : 0x08520100
+ *      1 : Media type (NAND / SDMC)
+ *      2 : Low word of the saveid to create
+ *      3 : High word of the saveid to create
+ *  Outputs:
+ *      1 : Result of function, 0 on success, otherwise error code
+ */
+static void DeleteExtSaveData(Service::Interface* self) {
+    // TODO(Subv): Figure out the other parameters.
```
Comment (bunnei) [inline src/core/hle/service/fs/fs_user.cpp]: What other params? (please document that there are unknown parameters in the docstring)

**Comment by bunnei** [comment]
Mostly LGTM

**Comment by Subv** [comment]
Updated

**Comment by archshift** [comment]
LGTM

**Pull request merged**